### PR TITLE
events/nats: Update consumer configuration when a consumer exists

### PR DESCRIPTION
**Pull request by joelrebel**
Updates a consumer configuration if a consumer exists.
Adds tests for consumer add/update.

### CODE REVIEW

File: events/nats.go [hunk 1]
```diff
@@ -218,6 +214,17 @@ func (n *NatsJetstream) addConsumer() error {
 		FilterSubject: n.parameters.Consumer.FilterSubject,
 	}
 
+	// Update consumer configuration when one exists
+	for name := range n.jsctx.ConsumerNames(n.parameters.Stream.Name) {
+		if name == n.parameters.Consumer.Name {
+			if _, err := n.jsctx.UpdateConsumer(n.parameters.Stream.Name, cfg); err != nil {
+				return errors.Wrap(ErrNatsJetstreamUpdateConsumer, err.Error())
```
Comment (DoctorVin) [inline events/nats.go:221]: Consider flipping this so the caller can test against the native `NATS` error?

File: events/nats.go [hunk 2]
```diff
@@ -218,6 +214,17 @@ func (n *NatsJetstream) addConsumer() error {
 		FilterSubject: n.parameters.Consumer.FilterSubject,
 	}
 
+	// Update consumer configuration when one exists
+	for name := range n.jsctx.ConsumerNames(n.parameters.Stream.Name) {
```
Comment (DoctorVin) [inline events/nats.go:218]: Are we opening ourselves up for a hassle when multiple controllers are restarted and attempt to update the consumer concurrently?

**Review by DoctorVin — COMMENTED** [review]
It looks OK, but I am concerned about some of the edge cases.

File: events/nats.go (hunk 2 above)
Comment (joelrebel) [reply events/nats.go:223]: with the latest change, it'll be a race between the workers to update the configuration so it should be ok ?

Comment (DoctorVin) [reply events/nats.go:223]: Probably, yes.

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**